### PR TITLE
Address single-threadedness in FAQ

### PR DESCRIPTION
**Pull request by MaryaBelanger**
Staged here (link to each section): 
- https://dart-dev-marya.web.app/faq#q-is-dart-single-threaded 
- https://dart-dev-marya.web.app/faq#q-is-dart-single-threaded-on-the-web

Part of #4042 

(also added a line to the isolates section of the language tour to explicitly mention multi-threading)

### CODE REVIEW

**Review by sigmundch — APPROVED** [review]
Thanks @MaryaBelanger! Looks great! I'll let @mit-mit give the final approval :)

File: src/faq.md [hunk 1]
```diff
@@ -202,6 +202,20 @@ Inside or outside of Google, [every Flutter app][FlutterShowcase] uses Dart.
 
 ## Native execution
 
+### Q. Is Dart single-threaded?
+
+No. On native targets, Dart features like isolates and async-await
```
Comment (domesticmouse) [inline src/faq.md:207]: Async/Await isn't threading. Argh.

**Review by domesticmouse — COMMENTED** [review]
We need to differentiate threads and concurrency. This is an area of mass confusion amongst developers. 

Relevant content: https://www.youtube.com/watch?v=oV9rvDllKEg

File: src/faq.md (hunk 1 above)
Comment (MaryaBelanger) [reply src/faq.md:207]: 😱 

Will address, thanks Brett!

File: src/_guides/language/language-tour.md [hunk 2]
```diff
@@ -4541,7 +4541,9 @@ concurrency is error prone and can lead to complicated code.
 
 Instead of threads, all Dart code runs inside of *isolates*. 
 Each Dart isolate has a single thread of execution and
-shares no mutable objects with other isolates.  
+shares no mutable objects with other isolates.
+Spinning up multiple isolates creates multiple threads of execution,
+enabling multi-threading without the typical issues that surround it.
```
Comment (atsansone) [inline src/_guides/language/language-tour.md:4546]: **suggestion:** 

```suggestion
Each Dart isolate uses one execution thread and
shares no mutable objects with other isolates.
Spinning up multiple isolates creates multiple threads of execution.
This enables multi-threading without the typical issues that surround it.
```

**question:** What issues?

**Review by atsansone — CHANGES_REQUESTED** [review]
I would have the issues I raised in the other PR resolved and merged first, then this one.

File: src/_guides/language/language-tour.md (hunk 2 above)
Comment (mit-mit) [reply src/_guides/language/language-tour.md:4546]: The primary issue is race conditions, e.g.: https://en.wikipedia.org/wiki/Race_condition#In_software

File: src/faq.md [hunk 3]
```diff
@@ -266,6 +280,21 @@ You can also use the [Flutter framework]({{site.flutter}}), which has [web suppo
 [No.]({{site.news}}/2015/03/dart-for-entire-web.html)
 Dart is designed to compile to JavaScript to run across the modern web.
 
+### Q. Is Dart single-threaded on the web?
+
+Dart is *mostly* single-threaded on the web. 
+Web apps can use concurrency features like async-await,
```
Comment (lrhn) [inline src/faq.md:286]: Async-await, futures and streams is not what I'd consider "Conccurency". Only web-workers actually allow concurrent execution.

File: src/_guides/language/language-tour.md (hunk 2 above)
Comment (MaryaBelanger) [reply src/_guides/language/language-tour.md:4553]: > Each Dart isolate uses one execution thread and

Going to keep "a single thread of execution" as is to match the language of the original issue.

File: src/_guides/language/language-tour.md [hunk 4]
```diff
@@ -4547,8 +4547,10 @@ shared-memory threads running concurrently. However, shared-state
 concurrency is error prone and can lead to complicated code.
 
 Instead of threads, all Dart code runs inside of *isolates*. 
-Each Dart isolate has a single thread of execution and
-shares no mutable objects with other isolates.  
+Each Dart isolate uses a single thread of execution and
+shares no mutable objects with other isolates.
+Spinning up multiple isolates creates multiple threads of execution.
+This enables multi-threading without its primary drawback, race conditions.
```
Comment (mit-mit) [inline src/_guides/language/language-tour.md:4553]: Should we link `race conditions` to the wikipedia page? I find their example pretty easy to read.

File: src/faq.md [hunk 5]
```diff
@@ -202,6 +202,20 @@ Inside or outside of Google, [every Flutter app][FlutterShowcase] uses Dart.
...
+No. On native targets, Dart features like isolates and async-await
+can enable multiple threads of execution at any given time.
+The Dart VM will utilize multiple cores of a multi-core processor
+to run those threads concurrently.
+
+[Dart's concurrency architecture](/guides/language/concurrency)
+abstracts the complex, error-prone code of typical shared-memory threading,
+which might explain the misconception that Dart is single-threaded.
+
+This is not quite the case for web apps, though; see
```
Comment (mit-mit) [inline src/faq.md:216]: It's a bit hard to grok what "this" binds back to here. So perhaps something like "Note that concurrency works differently in Dart web apps; ..."

**Review by mit-mit — COMMENTED** [review]
This is shaping up nicely!

**Review by mit-mit — APPROVED** [review]

**Pull request merged**